se2_planning/include/se2_planning/GridMapLazyStateValidator.hpp

```diff
@@ -23,32 +25,56 @@ class GridMapLazyStateValidator : public GridMapStateValidator {
   ~GridMapLazyStateValidator() override = default;
 
   bool isStateValid(const State& state) const final;
-  void initialize() final;
+  void initialize() override;
   bool isInitialized() const final;
 
   void setIsUseRandomizedStrategy(bool value);
   bool getIsUseRandomizedStrategy() const;
 
   void setIsUseEarlyStoppingHeuristic(bool value);
-  bool setIsUseEarlyStoppingHeuristic() const;
+  bool getIsUseEarlyStoppingHeuristic() const;
 
   void setSeed(int value);
   int getSeed() const;
 
+  void setStateValidityCheckingMethod(StateValidityCheckingMethod value);
+  StateValidityCheckingMethod getStateValidityCheckingMethod() const;
+
+  void setStateValidityThreshold(double value);
```

Comment:
we need to explain what is that threshold. e.g. cell value < threshold is considered valid.